nntrainer/tensor/tensor.cpp

```diff
@@ -2644,6 +2668,51 @@ void Tensor::print(std::ostream &out) const {
 #else
     throw std::invalid_argument("Error: enable-fp16 is not enabled");
 #endif
+  } else if (getDataType() == ml::train::TensorDim::DataType::QINT8) {
+    const int8_t *data = getData<int8_t>();
+    unsigned int len = size();
+    out << "data addr: " << (float *)data << '\n';
```

Comment:
Avoid using C-style casts ```(float *)data``` Instead, ```reinterpret_cast<float*>(data)``` to make the cast explicit and avoid potential issues with type safety.
```suggestion
    out << "data addr: " << reinterpret_cast<const float*>(data) << '\n';
```

Reply:
Applied!